src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -243,11 +243,20 @@ public RestoreCommand(FrameworkName fallbackFramework)
             var projectLockFilePath = Path.Combine(projectFolder, LockFileFormat.LockFileName);
 
             Runtime.Project project;
-            if (!Runtime.Project.TryGetProject(projectJsonPath, out project))
+            var diagnostics = new List<ICompilationMessage>();
+            if (!Runtime.Project.TryGetProject(projectJsonPath, out project, diagnostics))
             {
                 throw new Exception("TODO: project.json parse error");
             }
 
+            if (diagnostics.HasErrors())
+            {
+                var exceptions = diagnostics
+                    .Where(x => x.Severity == CompilationMessageSeverity.Error)
+                    .Select(x => new Exception(x.FormattedMessage));
+                throw new AggregateException(exceptions);
```

Comment:
Don't throw exceptions from command line tools. It looks bad :smile: . I'm also not sure if we should fail up front here. Maybe we can just warn about unknown frameworks during restore.